docs/discussion/selection-api.md

```diff
@@ -0,0 +1,378 @@
+[Table](https://docs.astropy.org/en/stable/table/index.html),
+[Region](https://astropy-regions.readthedocs.io/en/stable/)).
+This provides a familiar and well-documented foundation for users
+working within the scientific Python ecosystem.
+
+#### User-Defined Catalogs
+
+All selection interfaces are designed with the assumption that catalogs
+are loaded into aladin-lite programmatically by the user within their
+Jupyter notebook. Catalogs added directly via the widget interface are
+not considered within the scope of these APIs, ensuring greater control
+and consistency for programmatic workflows.
+
+## 1.2. Feature Requirements
+
+#### Selection Requirements
+
+As a notebook user/scientist I want to\...
+
+- slect a source within a catalog using a programmatic API
+
+- select a list of sources within all loaded catalogs using a
+  programmatic API
+
+- select all sources within a defined region using a programmatic API
+
+- select a footprint using a programmatic API 
+
+#### Data Retrieval Requirements 
+
+As a notebook user/scientist I want to\...
+
+- access the list of currently selected sources using a programmatic
+  API 
+
+- access the region of currently selected sources, if one was used to
+  select the sources, using a programmatic API 
+
+- access the footprint currently selected using a programmatic API 
+
+#### Listener Requirements 
+
+As a developer, I want to\...
+
+- listen for changes to the selected sources 
+
+- listen for changes to the selection region
+
+- listen for changes to the selected footprint
+
+#### Misc. Requirements
+
+As a notebook user/scientist I want to\...
+
+- be able to remove the current selection using a programmatic API
+
+## 1.3. Example Use Cases
+
+**Use case 1:** A scientist is interested in all sources in a catalog
+with a redshift within a defined range. The query might look like the
+following
+```
+aladin.selectSources(match_func: lambda source: source.rshift > 1 and
+source.rshift < 2)
+```
+**Use case 2:** A scientist is interested in a subset of the sources in
+a catalog. Since they loaded the catalog themselves, they know that
+there is an objId  field that uniquely identifies the source. The query
+might look like the following 
+```
+aladin.selectSources(sources=catalog_subset, match_key="objId")
+```
+maybe they instead want to use RA and DEC to identify the source instead
+```
+aladin.selectSources(sources=catalog_subset, match_keys=["ra",
+"dec"])
+```
+**Use case 3:** A software engineer wants to create an interoperable
+layer between an aladin-lite widget and another visualization tool, such
+as [jdaviz](https://jdaviz.readthedocs.io/en/stable/). They would then
+need to use a combination of the APIs listed in this doc to ensure that
+selection in aladin-lite are reflected in jdaviz and vice versa. 
+
+# 2. ipyaladin interfaces
+
+## 2.1. GET
+
+#### GetSelection
+```
+def GetSelection() -> Table
+```
+Returns an Astropy table containing the currently selected catalog
```

Comment:
I think we should spell out in a little more detail how we handle the case of multiple catalog (or footprint) layers.  The current behavior in Aladin Lite seems intuitive in that a selection action applies to all layers (at least visible ones, so visibility may be another complication for the programmatic API).  Then GetSelections returns a list-like structure (one item per layer) of lists/tables describing the selected objects for each layer.  This would apply to the Python and JavaScript APIs.